cmd/manager/resultcollector.go

```diff
@@ -178,52 +180,72 @@ func waitForResultsFile(filename string, timeout int64) *os.File {
 	return nil
 }
 
-func resultNeedsCompression(contents []byte) bool {
-	return len(contents) > 1048570
+func resultNeedsCompression(size int64) bool {
+	return size > 1048570
 }
 
-func compressResults(contents []byte) ([]byte, error) {
+func compressResults(contents io.Reader) (io.Reader, error) {
 	// Encode the contents ascii, compress it with gzip, b64encode it so it
 	// can be stored in the configmap.
 	var buffer bytes.Buffer
 	w, err := bzip2.NewWriter(&buffer, &bzip2.WriterConfig{Level: bzip2.BestCompression})
 	if err != nil {
 		return nil, err
 	}
-	w.Write([]byte(contents))
-	w.Close()
-	return buffer.Bytes(), nil
+	_, err = io.Copy(w, contents)
+	if err != nil {
+		return nil, err
+	}
+	defer w.Close()
```

Comment:
This reads as if copy fails, then we never call the defer and never close the writer..

Reply:
Good catch. Fixed